flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -597,14 +676,30 @@ void testOnLeaderInformationChangeIsIgnoredAfterLeaderElectionBeingStop() throws
                         () -> {
                             grantLeadership();
 
-                            assertThat(storedLeaderInformation.get().forContenderID(contenderID))
-                                    .isPresent();
+                            applyToBothContenderContexts(
+                                    ctx ->
+                                            assertThat(
+                                                            storedLeaderInformation
+                                                                    .get()
+                                                                    .forContenderID(
+                                                                            ctx.contenderID))
+                                                    .isPresent());
+
+                            contenderContext0.leaderElection.close();
```

Comment:
why isn't this done within `applyToBothContenderContexts`? it runs the same logic for both contenders, no?

Reply:
Because it checked also the other contender which the test expects to be corrected after the first close call. But I noticed that this isn't the right behavior: We shouldn't touch the LeaderInformation for a contender that wasn't mentioned in the `notifyLeaderInformationChange` call. I have to revisit the behavior of the leaderinformation change event handling